gcloud-java-compute/src/main/java/com/google/gcloud/compute/DiskTypeId.java

```diff
@@ -61,25 +63,40 @@ public String diskType() {
     return diskType;
   }
 
+  /**
+   * Returns the name of the zone this disk type belongs to.
+   */
+  public String zone() {
+    return zone;
+  }
+
+  /**
+   * Returns the identity of the zone this disk type belongs to.
+   */
+  public ZoneId zoneId() {
+    return ZoneId.of(project(), zone);
+  }
+
   @Override
   public String selfLink() {
-    return super.selfLink() + "/diskTypes/" + diskType;
+    return super.selfLink() + "/zones/" + zone + "/diskTypes/" + diskType;
   }
 
   @Override
   MoreObjects.ToStringHelper toStringHelper() {
-    return super.toStringHelper().add("diskType", diskType);
+    return super.toStringHelper().add("zone", zone).add("diskType", diskType);
   }
 
   @Override
   public int hashCode() {
-    return Objects.hash(super.baseHashCode(), diskType);
+    return Objects.hash(super.baseHashCode(), zone, diskType);
   }
 
   @Override
   public boolean equals(Object obj) {
     return obj instanceof DiskTypeId
         && baseEquals((DiskTypeId) obj)
+        && Objects.equals(zone, ((DiskTypeId) obj).zone)
```

Comment:
Nit, with that many casts I would have kept it in local variable. Fine as is though.